app/models/container_node.rb

```diff
@@ -77,6 +77,14 @@ def ipaddress
   end
 
   def cockpit_url
-    URI::HTTP.build(:host => ipaddress, :port => 9090)
+    address = ipaddress || name
+    if MiqRegion.my_region.role_active?("cockpit_ws")
+      uri = URI(MiqCockpit::WS::MIQ_REDIRECT_URL)
+      uri.query = {:ip => address }.to_query
```

Comment:
This one is different, the ip here is in the query portion of the url, not the hostname.